src/nnet3/nnet-nnet.cc

```diff
@@ -665,14 +665,29 @@ void Nnet::Check() const {
     const NetworkNode &node = nodes_[n];
     std::string node_name = node_names_[n];
     KALDI_ASSERT(GetNodeIndex(node_name) == n);
+    int32 node_index_reserved = GetNodeIndex(node_name +
+                                             "_STATE_PREVIOUS_MINIBATCH");
+    if (node_index_reserved != -1 && 
+        (!IsInputNode(node_index_reserved) || !IsOutputNode(n) ||
+      node_name == "output"))
+      KALDI_ERR << " The name \"" << node_name << "\" with suffix \""
```

Comment:
I don't like the fact that this type of code, which relates to a specific use of the core nnet code, is mixed up in the 'core' parts of the framework.  None of this stuff should affect the core parts of the framework at all.  If you want to check this kind of thing, you can create a function that checks that the names have been correctly assigned, and call it from the code that deals with this type of framework.  See the function IsSimpleNnet as an example-- only the code that cares that the nnet has this particular configuration, checks it.